### ISSUE TITLE
adc~ with external usb soundcard not working

### ISSUE THREAD

**Issue by cotik1** [danomatika/PdParty#107]
Hello Dan,
thank you for the amazing PdPdrty!
I am experiencing a weird problem with adc~.
It works fine with the iternal ipad soundcard, but as soon as I switch to a USB one (Auident iD14 mkII), the adc~ stops working (no incoming audio),
and also signals routed to dac~ outs are routed back into the adc~ --weird.
thought it might be a soundcard thing, but it works just fine in Mobmuplat and AUM..
the same happens in my Iphone when I plug in the soundcard
any ideas?
I could send you some screen grabs if you think it might help
all the best,
Coti

**Comment by danomatika** [comment]
Need info: what device, what iOS version, which PdParty version, etc.

**Comment by cotik1** [comment]
tried it (with same problem) on 2 devices:
iPad pro 11inch 4th gen, ipadOS 16.5, PdParty 1.4.0 - connected to soundcard through usb-c to usb 3 hub
iPhone Xs, iOS 16.3.1, PdParty 1.3.2 - connected to soundcard through lightning to USB camera adapter+ powered USB hub.
soundcard Audient iD14 mkII
patch created on Pd version 0.54.0
need more info?
thanks!

**Comment by danomatika** [comment]
Hmm strange. I mainly use PdParty with a USB audio interface via a hub and have no issues. What is in your patch... do you use `[soundinput]` instead of `[adc~]` as required by PdParty, and noted in the user guide?

**Comment by cotik1** [comment]
hello again,
I was indeed using the adc~ and dac~, bur swapping them for [sundinput] and [soundoutput] made no difference.
I created this super simple patch to troubleshoot the problem.
here it is:

#N canvas 428 572 394 417 12;
#X obj 45 66 env~;
#X obj 45 90 s \$0-lvl_1;
#X obj 49 129 hsl 100 19 0 100 0 0 empty \$0-lvl_1 empty -2 -10 0 12 #fcfcfc #000000 #000000 0 1;
#X obj 148 65 env~;
#X obj 48 159 hsl 100 19 0 100 0 0 empty \$0-lvl_2 empty -2 -10 0 12 #fcfcfc #000000 #000000 0 1;
#X obj 148 89 s \$0-lvl_2;
#X obj 178 140 tgl 19 0 \$0-test-s empty test_L 0 -10 0 12 #fcfcfc #000000 #000000 0 1;
#X obj 178 164 r \$0-test-s;
#X msg 178 188 \$1 200;
#X obj 178 212 line~;
#X obj 178 282 *~;
#X obj 278 140 tgl 19 0 \$0-test2-s empty test_R 0 -10 0 12 #fcfcfc #000000 #000000 0 1;
#X msg 278 188 \$1 200;
#X obj 278 212 line~;
#X obj 281 282 *~;
#X obj 278 164 r \$0-test2-s;
#X obj 178 309 soundoutput;
#X obj 45 19 soundinput;
#X obj 210 251 osc~ 220;
#X text 10 128 in 1;
#X text 10 162 in 2;
#X connect 0 0 1 0;
#X connect 3 0 5 0;
#X connect 7 0 8 0;
#X connect 8 0 9 0;
#X connect 9 0 10 1;
#X connect 10 0 16 0;
#X connect 12 0 13 0;
#X connect 13 0 14 1;
#X connect 14 0 16 1;
#X connect 15 0 12 0;
#X connect 17 0 0 0;
#X connect 17 1 3 0;
#X connect 18 0 14 0;
#X connect 18 0 10 0;

**Comment by danomatika** [comment]
I'm busy at the moment, but I can say there are some major problems with the patch which keep it from working. I can send you a working version later.

**Comment by cotik1** [comment]
Thank you!

**Comment by danomatika** [comment]
Try this:

[test.pd.zip](https://github.com/danomatika/PdParty/files/13325975/test.pd.zip)

**Comment by cotik1** [comment]
Thank you Dan!
But apart from you fixing the [osc~] missing tilda typo, nothing much has
changed…
Here is a video showing that weird returning signal from the outs to the
ins…
(Please Ignore my daughter playing piano in the background)

https://www.dropbox.com/scl/fi/k34iglhndv9z4f757kfz5/feedback.MOV?rlkey=p7pbz22kavbd41c58scqmqjd4&dl=0

To me it seems some kind of driver weirdness..

Coti Kappa
cotik.com
+306973050696

**Comment by cotik1** [comment]
Thank you Dan for your time!
still trying to figure out why it doesen't work (and both MobMuplat and Aum do)
can I ask what USB audio interface are you using? thinking of borrowing one and making some tests with it if I can..

**Comment by danomatika** [comment]
I have a pair of old Roland Edirol UA-25 and UA-25 EXs. They are USB 1.1 stereo audio interfaces with XLR preamps and work with anything.

PdParty doesn't do anything special with the audio handling on iOS. It just opens a normal RemoteIO audio unit. Did you try any of the examples or tests which comes with PdParty before deciding your setup doesn't work?

**Comment by danomatika** [comment]
> But apart from you fixing the [osc~] missing tilda typo, nothing much has

The other major problems were the output lines were always going to be 0 as they received a `[0 200<` no matter what output the toggles send *and* the output multipliers are not at a the signal rate, ie. `[* ]` should be `[*~]`.

**Comment by danomatika** [comment]
@cotik1 Can you check the `tests/multi_channel.pd` patch? Also, due to a copy/paste error with your original patch, I found a  potential crash bug. You'll get tangential credit in the changelog. ;)

@Josef-N Do you notice any issued with multi-channel audio devices, ie. more than stereo?

**Comment by Josef-N** [comment]
I did not find any issue with my audio interface Keith McMillen K-Mix, 8 In, 10 out.
Tested with PdParty 1.4.1 on iPad Pro, 11", 3rd Gen, OS 17.1.1
I am using dac~ and adc~. Output only stereo, all gain controls are working. In my patch inputs are selectable, 1–8, this works also. 
This was a good point for me to test, because I have been using patches that were created with earlier versions of Pd. I tried those first, then opened and re-saved them in current Pd 0.54-1, both versions are working.
I was curious, because I am using sigmund~ for the input signal, and there has been a change in Pd versions. Good to know, that it does not affect my patches.

**Comment by cotik1** [comment]
hello, tired the tests/multi_channel.pd, same problem as before, no sound input, out feeding back to in.
I was convinced it was a hardware/software combination, so, as I like so much PdParty, I went and bough a NI Komplete audio 6 card, and it works fine. Of course  we still don't know why the Audient works with all other software but not PdParty.. but maybe we will find out down the line... thank you so much for your time!!

**Comment by danomatika** [comment]
Sorry, I have no answer without having the device to try myself.

**Comment by cotik1** [comment]
Totally understand, Thank you Dan!!

**Comment by danomatika** [comment]
That being said (written?), if you send me the device, I could try debugging. 😄 On loan, of course.

**Comment by cotik1** [comment]
I actually thought of that... but I am using it on a quite regular basis..  (and the NI one is now being used by my drummer)
let me think how I can do that... what country are you in? Germany?

**Comment by danomatika** [comment]
Yeah, I'm in Karlsruhe, DE. If you're in Germany, DHL Paket with tracking is probably best. Send me an email, my address is in the readme.

**Comment by cotik1** [comment]
I'm in Athens, Greece, will see if I can borrow another card for a few weeks, and will check postage costs.
thanks again!

**Comment by danomatika** [comment]
Closing due to inactivity
